### DISCUSSION TITLE
[Select.Root] pass value as object

### DISCUSSION THREAD

**Discussion by L-Blondy** [radix-ui/primitives#2527] [Help]
Hi there

I see in the docs of the Select peimitive that that root component accepts only values of type string. Indeed typescript forces it too.

Passing the value as an object seems to work so far in my project.

Is there any technical reason why I should not use objects as values?

**Comment by benoitgrelard** [comment]
It should work the same way as a native select, which only accepts strings.
I suspect passing objects would have issues in form serialisations potentially.
